### PR TITLE
test/e2e/: use framework.Equal() replace gomega.Expect(...).To(gomega…

### PR DESCRIPTION
**Pull request by tanjunchen**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

use framework.Equal(sth,True,explain...) replace gomega.Expect(sth).To(gomega.BeTrue(),explain...)
and then we can remove gomega in some folders

**Which issue(s) this PR fixes**:

 
Ref: #79686

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
None
```

### CODE REVIEW

**Review by oomichi — COMMENTED** [review]
Thanks for doing this.

/lgtm

**Comment by tanjunchen** [comment]
@oomichi  i push it again due to conflict

**Review by oomichi — COMMENTED** [review]
/lgtm

File: test/e2e/apimachinery/webhook.go [hunk 1]
```diff
@@ -995,7 +995,7 @@ func testMutatingConfigMapWebhook(f *framework.Framework) {
 	client := f.ClientSet
 	configMap := toBeMutatedConfigMap(f)
 	mutatedConfigMap, err := client.CoreV1().ConfigMaps(f.Namespace.Name).Create(configMap)
-	gomega.Expect(err).To(gomega.BeNil())
+	framework.ExpectNoError(err)
```
Comment (alejandrox1) [inline test/e2e/apimachinery/webhook.go]: Nice catch!

**Review by alejandrox1 — COMMENTED** [review]
Thank you for working on this @tanjunchen !
/approve
/priority backlog
/milestone v1.18

**Pull request merged**